CMakeLists.txt

```
@@ -1,33 +1,40 @@
 # @copyright See `LICENSE` in the root directory of this project.
 
-cmake_minimum_required(VERSION 3.11) # If ryml >= 0.5.0 is included via FetchContent or submodule, this minimum is 3.12.
+cmake_minimum_required(VERSION 3.10)
 
 project("ANTLER Project Tools " VERSION 1.0.0)
 
 include( ./common.cmake REQUIRED )
 
+
+
+option(ANTLER_PROJ_PEDANTIC_BUILD "Pedantic builds" On)
+option(ANTLER_PROJ_BUILD_TESTS "Build and run the tests." On)
+option(ANTLER_PROJ_TIDY_CHECKS "Uses clang-tidy in building." On)
+
 add_subdirectory( external )
 add_subdirectory( include )
 add_subdirectory( src )
 add_subdirectory( tools )
 
-
-option(PEDANTIC_BUILD "Pedantic builds" On)
-option(BUILD_TESTS "Build and run the tests." On)
-
-if(BUILD_TESTS)
+if(ANTLER_PROJ_BUILD_TESTS)
   enable_testing()
   add_subdirectory( tests )
 endif()
 
+# Clang-tidy support
+if (ANTLER_PROJ_TIDY_CHECKS)
+  find_program(CLANG_TIDY_EXE NAMES "clang-tidy" REQUIRED)
+  set(CLANG_TIDY_COMMAND "${CLANG_TIDY_EXE}" "-checks=*,modernize-*")
+endif()
 
 # Copy licenses to BUILD dir
 #
 #  NOTE: This is used due to legacy requirements instead of:
 #    `install(FILE <license files> COMPONENT Documentation DESTINATION share/aproj/licenses)`
-configure_file(LICENSE ${CMAKE_BINARY_DIR}/licenses/LICENSE COPYONLY)
+configure_file(LICENSE ${CMAKE_CURRENT_BINARY_DIR}/licenses/LICENSE COPYONLY)
```

I am beginning to think we should create a `copy_license()` function. But in the meantime, I suggest we make these more readable. 
```suggestion
configure_file(LICENSE ./licenses/ COPYONLY)
```

---

see 1afe6e5f11b6f879d182ae792838066e6e3f96e7

---

`CMAKE_CURRENT_BINARY_DIR` is something I added to allow it to be submoduled. I'm not sure about outcome of your suggested change but the goal is to have it in a current configured binary directory